rse-cmdline-git.Rmd

```diff
@@ -0,0 +1,3536 @@
+[BitBucket](https://bitbucket.org/),
+[GitLab](https://gitlab.com/) or
+another Git host server
+in a later lesson will include this information.
+
+For these lessons, we will be interacting with [GitHub](https://github.com/) and
+so the email address used should be the same as the one used when setting up your GitHub account.
+If you are concerned about privacy,
+please review [GitHub's instructions for keeping your email address private][git-privacy]. 
+
+Dracula also has to set his favorite text editor, following this table:
+
+| Editor             | Configuration command                            |
+|:-------------------|:-------------------------------------------------|
+| Atom | `$ git config --global core.editor "atom --wait"`|
+| nano               | `$ git config --global core.editor "nano -w"`    |
+| BBEdit (Mac, with command line tools) | `$ git config --global core.editor "bbedit -w"`    |
+| Sublime Text (Mac) | `$ git config --global core.editor "/Applications/Sublime\ Text.app/Contents/SharedSupport/bin/subl -n -w"` |
+| Sublime Text (Win, 32-bit install) | `$ git config --global core.editor "'c:/program files (x86)/sublime text 3/sublime_text.exe' -w"` |
+| Sublime Text (Win, 64-bit install) | `$ git config --global core.editor "'c:/program files/sublime text 3/sublime_text.exe' -w"` |
+| Notepad++ (Win, 32-bit install)    | `$ git config --global core.editor "'c:/program files (x86)/Notepad++/notepad++.exe' -multiInst -notabbar -nosession -noPlugin"`|
+| Notepad++ (Win, 64-bit install)    | `$ git config --global core.editor "'c:/program files/Notepad++/notepad++.exe' -multiInst -notabbar -nosession -noPlugin"`|
+| Kate (Linux)       | `$ git config --global core.editor "kate"`       |
+| Gedit (Linux)      | `$ git config --global core.editor "gedit --wait --new-window"`   |
+| Scratch (Linux)       | `$ git config --global core.editor "scratch-text-editor"`  |
+| Emacs              | `$ git config --global core.editor "emacs"`   |
+| Vim                | `$ git config --global core.editor "vim"`   |
+| VS Code                | `$ git config --global core.editor "code --wait"`   |
+
+It is possible to reconfigure the text editor for Git whenever you want to change it.
+
+> **Exiting Vim**
+>
+> Note that Vim is the default editor for many programs.
+> If you haven't used Vim before and wish to exit a session
+> without saving your changes, press <kbd>Esc</kbd> then type `:q!` and hit <kbd>Return</kbd>.
+> If you want to save your changes and quit, press <kbd>Esc</kbd> then type `:wq` and hit <kbd>Return</kbd>.
+
+The four commands we just ran above only need to be run once: the flag `--global` tells Git
+to use the settings for every project, in your user account, on this computer.
+
+You can check your settings at any time:
+
+```shell
+$ git config --list
+```
+
+You can change your configuration as many times as you want: use the
+same commands to choose another editor or update your email address.
+
+> **Proxy**
```

Comment:
How much do we want to add "side material" in the main text of the course material? And also, if this course is targeted to novices, this will be confusing (proxies confuse me too).